common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -632,6 +737,14 @@ public void registerExecutor(String appId, ExecutorShuffleInfo executorInfo) {
           appsShuffleInfo.compute(appId, (id, appShuffleInfo) -> {
             if (appShuffleInfo == null || attemptId > appShuffleInfo.attemptId) {
               originalAppShuffleInfo.set(appShuffleInfo);
+              AppPathsInfo appPathsInfo = new AppPathsInfo(appId, executorInfo.localDirs,
+                  mergeDir, executorInfo.subDirsPerLocalDir);
+              // Clean up the outdated App Attempt local path info in the DB and
+              // put the newly registered local path info from newer attempt into the DB.
+              if (appShuffleInfo != null) {
+                removeAppAttemptPathInfoFromDB(new AppAttemptId(appId, appShuffleInfo.attemptId));
+              }
+              writeAppPathsInfoToDb(appId, attemptId, appPathsInfo);
```

Comment:
The goal is to save the latest attempt of an application. In that case, we need to make sure if the delete from db fails, we don't ignore it and write the next attempt. Right now, `removeAppAttemptPathInfoFromDB` swallows the exception so it can happen that we have multiple entries in the db.

Reply:
Umm, so we should better still adding the multiple attempts handling during DB reloading? If we have to guarantee the success of DB removal, but it fails, should we fail the new app attempt Executor registration here? I feel the later one is a little bit overkill. WDYT?

Reply:
Updated as suggested.